packages/client/hmi-client/src/components/model-template/tera-model-template-editor.vue

```diff
@@ -108,29 +119,49 @@ import type {
 	ModelTemplateJunction
 } from '@/types/model-templating';
 import * as modelTemplatingService from '@/services/model-templating';
+import SelectButton from 'primevue/selectbutton';
+import { KernelSessionManager } from '@/services/jupyter';
+import { logger } from '@/utils/logger';
 import TeraInfiniteCanvas from '../widgets/tera-infinite-canvas.vue';
 import TeraModelTemplate from './tera-model-template.vue';
 import TeraModelJunction from './tera-model-junction.vue';
 import TeraCanvasItem from '../widgets/tera-canvas-item.vue';
 
-defineProps<{
+const props = defineProps<{
 	model?: Model;
 }>();
 
+enum EditorView {
+	Template = 'Template',
+	Flat = 'Flat'
+}
```

Comment:
It isn't so much that we have template-view vs flat-view; they are both templates, just one is flat and one is not. 

Maybe it is better to call them `Decomposed` and `Flattened` instead? Any thoughts?

Also you are calling the enum EditorView, but you are using them where we are setting "formats"....so perhaps we should rename the enum to EditorFormat to match, or vice versa.

Reply:
Yeah, those enum names make sense. The designs just had them as `Template` and `Flat`. Will apply these.